server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/method/IdentitySetDeletePerformer.scala

```diff
@@ -0,0 +1,75 @@
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.jmap.method
+
+import eu.timepit.refined.auto._
+import javax.inject.Inject
+import org.apache.james.jmap.api.identity.{IdentityForbiddenDeleteException, IdentityRepository}
+import org.apache.james.jmap.api.model.IdentityId
+import org.apache.james.jmap.core.{Properties, SetError}
+import org.apache.james.jmap.core.SetError.SetErrorDescription
+import org.apache.james.jmap.mail.{IdentitySetRequest, UnparsedIdentityId}
+import org.apache.james.jmap.method.IdentitySetDeletePerformer.{IdentityDeletionFailure, IdentityDeletionResult, IdentityDeletionResults, IdentityDeletionSuccess}
+import org.apache.james.mailbox.MailboxSession
+import reactor.core.scala.publisher.{SFlux, SMono}
+import reactor.core.scheduler.Schedulers
+
+object IdentitySetDeletePerformer {
+  sealed trait IdentityDeletionResult
+  case class IdentityDeletionSuccess(identityId: IdentityId) extends IdentityDeletionResult
+  case class IdentityDeletionFailure(identityId: UnparsedIdentityId, exception: Throwable) extends IdentityDeletionResult {
+    def asIdentitySetError: SetError = exception match {
+      case e: IdentityForbiddenDeleteException => SetError.forbidden(SetErrorDescription(e.getMessage), Properties("id"))
+      case e: IllegalArgumentException => SetError.invalidArguments(SetErrorDescription(s"${identityId.id} is not a IdentityId: ${e.getMessage}"))
+      case _ => SetError.serverFail(SetErrorDescription(exception.getMessage))
+    }
+  }
+  case class IdentityDeletionResults(results: Seq[IdentityDeletionResult]) {
+    def destroyed: Seq[IdentityId] =
+      results.flatMap(result => result match {
+        case success: IdentityDeletionSuccess => Some(success)
+        case _ => None
+      }).map(_.identityId)
+
+    def retrieveErrors: Map[UnparsedIdentityId, SetError] =
+      results.flatMap(result => result match {
+        case failure: IdentityDeletionFailure => Some(failure.identityId, failure.asIdentitySetError)
+        case _ => None
+      })
+        .toMap
+  }
+}
+
+class IdentitySetDeletePerformer @Inject()(identityRepository: IdentityRepository) {
+  def destroy(identitySetRequest: IdentitySetRequest, mailboxSession: MailboxSession): SMono[IdentityDeletionResults] =
+    SFlux.fromIterable(identitySetRequest.destroy.getOrElse(Seq()))
+      .flatMap(unparsedId => delete(unparsedId, mailboxSession)
+        .onErrorRecover(e => IdentityDeletionFailure(unparsedId, e)),
+        maxConcurrency = 5)
+      .collectSeq()
+      .map(IdentityDeletionResults)
+
+  private def delete(unparsedId: UnparsedIdentityId, mailboxSession: MailboxSession): SMono[IdentityDeletionResult] =
+    unparsedId.validate
+      .fold(e => SMono.error(e),
+        id => SMono.fromPublisher(identityRepository.delete(mailboxSession.getUser, Seq(id)))
+          .subscribeOn(Schedulers.elastic())
```

Comment:
Why do we need an elastic scheduler here? This detail  should be handled within `identityRepository.delete`